probe-rs/src/config/chip_family.rs

```diff
@@ -13,17 +13,13 @@ pub struct ChipFamily {
     /// E.g. `nRF52832`.
     pub name: Cow<'static, str>,
     /// The JEP106 code of the manufacturer.
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub manufacturer: Option<JEP106Code>,
+    pub vendor: JEP106Code,
```

Comment:
Yes, the vendor can be extracted from the packs, which is why i made this change. Also, some smaller vendor might not be registered with arm, which is no issue since the jep106(0,0) code will always mean unknown manufacturer. i renamed the value to be more in line with what terminology most folks use.

Reply:
I don't like the usage of JEP106Code(0,0) to specify unknown manufacturer, I think it goes against common usage in Rust, and also makes the target specification files a bit confusing. Reading `manufacturer: 0` and `part: 0` seems confusing to me.

I also see a problem when comparing the values from the ROM table with the stored values, because the ROM table would contain the actual manufacturer, while the specification contains the "wrong" values.